plugins/lsp/default.nix

```diff
@@ -256,8 +256,10 @@ lib.nixvim.plugins.mkNeovimPlugin {
                 options = vim.tbl_extend("keep", options, __setup)
```

Comment:
I believe we can simplify this slightly, since multiple calls to `vim.lsp.config()` will use a table deep extend internally we can probably just call `vim.lsp.config()` multiple times.

Reply:
I think we can simplify to something like:

```lua
local name = server
local options = nil
if type(name) != 'string' then
  name = server.name
  options = server.extraOptions
end

vim.lsp.enable(name)
vim.lsp.config(name, __setup)
if options != nil then
  vim.lsp.config(name, options)
end
```